st2client/tests/unit/test_formatters.py

```diff
@@ -112,6 +115,12 @@ def test_execution_get_attributes(self):
         content = self._get_execution(argv)
         self.assertEqual(content, FIXTURES['results']['execution_get_attributes.txt'])
 
+    def test_execution_get_attribute_with_hours_in_elapsed_time(self):
+        argv = ['execution', 'get', DURATION_HOURS['id'], '--attr', 'status']
+        content = self._get_execution(argv)
```

Comment:
I think using `self._get_execution()` is the source of the error because `_get_execution()` has a mock so it always returns the json from `EXECUTION`, but you need it to return `DURATION_HOURS`.

Here's `_get_execution` with the mocked `EXECUTION` response:
```
    @mock.patch.object(
        httpclient.HTTPClient,
        "get",
        mock.MagicMock(
            return_value=base.FakeResponse(json.dumps(EXECUTION), 200, "OK", {})
        ),
    )
    def _get_execution(self, argv):
        self.assertEqual(self.shell.run(argv), 0)
        self._undo_console_redirect()
        with open(self.path, "r") as fd:
            content = fd.read()

        return content
```

That's why your example unit test that uses mock.MagicMock worked better than `self._get_execution()`.